src/main/java/org/entando/entando/plugins/jpseo/web/page/SeoPageController.java

```diff
@@ -0,0 +1,137 @@
+
+    @Autowired
+    private SeoPageValidator seoPageValidator;
+
+    @Autowired
+    private PageAuthorizationService authorizationService;
+
+    public IPageService getPageService() {
+        return pageService;
+    }
+
+    public void setPageService(SeoPageService pageService) {
+        this.pageService = pageService;
+    }
+
+    public SeoPageValidator getSeoPageValidator() {
+        return seoPageValidator;
+    }
+
+    public void setSeoPageValidator(SeoPageValidator seoPageValidator) {
+        this.seoPageValidator = seoPageValidator;
+    }
+
+    public PageAuthorizationService getAuthorizationService() {
+        return authorizationService;
+    }
+
+    public void setAuthorizationService(PageAuthorizationService authorizationService) {
+        this.authorizationService = authorizationService;
+    }
+
+    @Override
+    public ResponseEntity<RestResponse<PageDto, Map<String, String>>> getSeoPage(UserDetails user, String pageCode,
+            String status) {
+        logger.debug("get seo page {}", pageCode);
+
+        Map<String, String> metadata = new HashMap<>();
+        if (!this.getAuthorizationService().isAuth(user, pageCode)) {
+            return new ResponseEntity<>(new RestResponse<>(new PageDto(), metadata), HttpStatus.UNAUTHORIZED);
+        }
+        SeoPageDto page = (SeoPageDto) this.getPageService().getPage(pageCode, status);
+        metadata.put("status", status);
+        return new ResponseEntity<>(new RestResponse<>(page, metadata), HttpStatus.OK);
+    }
+
+    @Override
+    public ResponseEntity<SimpleRestResponse<SeoPageDto>> addPage(UserDetails user,
+            SeoPageRequest pageRequest, BindingResult bindingResult) throws ApsSystemException {
+        logger.debug("creating page with request {}", pageRequest);
+        //field validations
+        if (bindingResult.hasErrors()) {
+            throw new ValidationGenericException(bindingResult);
+        }
+
+        getSeoPageValidator().checkFriendlyCode(pageRequest.getSeoData().getFriendlyCode());
+
+        //business validations
+        getSeoPageValidator().validate(pageRequest, bindingResult);
+        if (bindingResult.hasErrors()) {
+            throw new ValidationConflictException(bindingResult);
+        }
+        SeoPageDto dto = (SeoPageDto) this.getPageService().addPage(pageRequest);
+        return new ResponseEntity<>(new SimpleRestResponse<>(dto), HttpStatus.OK);
+    }
+
+    @Override
+    public ResponseEntity<RestResponse<SeoPageDto, Map<String, String>>> updatePage(
+            UserDetails user,String pageCode,
+           SeoPageRequest pageRequest, BindingResult bindingResult) {
+        logger.debug("updating page {} with request {}", pageCode, pageRequest);
+
+        if (!this.getAuthorizationService().isAuth(user, pageCode)) {
+            throw new ResourcePermissionsException(bindingResult, user.getUsername(), pageCode);
+        }
+        //field validations
+        if (bindingResult.hasErrors()) {
```

Comment:
Would if you put all the validation in the seoPageValidator and you create a separate method for each of the methods?
Like `seoPageValidator().validateUpdatePage()`